tests/Unit/Controllers/Api/ApiVoucherControllerTest.php

```diff
@@ -0,0 +1,156 @@
+        $this->actingAs($this->user, 'api')
+            ->json('POST', $route, $data)
+            ->assertStatus(200)
+        ;
+        // There should be no token for this request
+        $this->assertEquals(0, StateToken::all()->count());
+
+        // Change the data for confirm
+        $data["transition"] = 'confirm';
+        // Resubmit
+        $this->actingAs($this->user, 'api')
+            ->json('POST', $route, $data)
+            ->assertStatus(200)
+        ;
+        // There should be a token for this request
+        $this->assertEquals(1, StateToken::all()->count());
+
+        // Age the StateToken;
+        $oldDate = Carbon::today()->subDays(31);
+        $oldVoucherStateToken = $oldVoucher
+            ->getPriorState()
+            ->stateToken()->first();
+        $oldVoucherStateToken->created_at = $oldDate;
+        $oldVoucherStateToken->save();
+        $oldVoucherStateToken->fresh();
+        $this->assertEquals($oldDate, $oldVoucherStateToken->created_at);
+
+        // Confirm the rest of the vouchers
+        $data = [
+            "trader_id" => 1,
+            "transition" => 'collect',
+            "vouchers" => $this->vouchers->pluck('code')->toArray()
+        ];
+        $route = route('api.voucher.transition');
+        $this->actingAs($this->user, 'api')
+            ->json('POST', $route, $data)
+            ->assertStatus(200)
+        ;
+        // Change the data for confirm
+        $data["transition"] = 'confirm';
+        // Resubmit
+        $this->actingAs($this->user, 'api')
+            ->json('POST', $route, $data)
+            ->assertStatus(200)
+        ;
+        // oldVoucher Should no longer have a token;
+        $oldVoucherStateToken = $oldVoucher
+            ->getPriorState()
+            ->stateToken()->first();
+        $this->assertNull($oldVoucherStateToken);
+
+        // There should be only 1 token for the second batch
+        $this->assertEquals(1, StateToken::all()->count());
+    }
+
+    /** @test */
+    public function testItAttachesTokensToPaymentPendingStates()
+    {
+        // Progress some vouchers to recorded state via the controller;
+        $data = [
+            "trader_id" => 1,
+            "transition" => 'collect',
+            "vouchers" => $this->vouchers->pluck('code')->toArray()
+        ];
+
+        $route = route('api.voucher.transition');
+
+        $this->actingAs($this->user, 'api')
+            ->json('POST', $route, $data)
+            ->assertStatus(200)
+        ;
+
+        // See there are no Tokens
+        $this->assertEquals(0, StateToken::all()->count());
+
+        // Change the data for confirm
+        $data["transition"] = 'confirm';
+
+        // Resubmit
+        $this->actingAs($this->user, 'api')
+            ->json('POST', $route, $data)
+            ->assertStatus(200)
+        ;
+
+        // There should be a token for this request
+        $this->assertEquals(1, StateToken::all()->count());
```

Comment:
Seems like you are repeating elements already tested in `testItTidiesOldTokensOnConfirmTransitions()`